Month3/LinkedList/LinkedList/TDLList.h

```diff
@@ -0,0 +1,404 @@
+		}
+
+		else if (Index == Size)
+		{
+			AddTail(NewElement);
+		}
+
+		else
+		{
+			FNode* NewNode = new FNode;
+			NewNode->Element = NewElement;
+			FNode* Current = Head;
+
+			for (int Currentindex = 0; Currentindex < Index; ++Currentindex)
+			{
+				Current = Current->Next;
+			}
+			NewNode->Next = Current;
+			NewNode->Prev = Current->Prev;
+			Current->Prev->Next = NewNode;
+			Current->Prev = NewNode;
+			Size++;
+		}
+
+
+
+	}
+
+
+
+	void RemoveTail()
+	{
+		FNode* Current = Tail;
+		Tail = Tail->Prev;
+		delete Current;
+		--Size;
+	}
+
+	void Remove(const int Index)
+	{
+		if (Index >= Size || Index < 0 || Size == 0)
+		{
+			return;
+		}
+
+		else if (Size == 1)
+		{
+			Clear();
+		}
+		else if (Index == (Size - 1))
+		{
+			RemoveTail();
+
+		}
+		else
+		{
+			FNode* Current = Head;
+
+			for (int CurrentIndex = 0; CurrentIndex < Index - 1; CurrentIndex++)
+			{
+				Current = Current->Next;
+			}
+
+			FNode* After = Current->Next;
+			Current->Next = After->Next;
+			After->Next = nullptr;
+			After->Prev = nullptr;
+			delete After;
+			Current->Next->Prev = Current;
+			Size--;
+		}
+
+	}
+
+
+
+	void Clear()
+	{
+		for (int CurrentIndex = 0; CurrentIndex < Size; ++CurrentIndex)
+		{
+			FNode* Current = Head;
+			Head = Head->Next;
+			delete Current;
+		}
+		Tail = nullptr;
+		Size = 0;
+	}
+
+	template<typename Pred>
+	void ForEach(const Pred& Predicate)
+	{
+		FNode* Current = Head;
+		for (int CurrentIndex = 0; CurrentIndex < Size; CurrentIndex++)
+		{
+			Predicate(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	template<typename Pred>
+
+	T& FindByPredicate(const Pred& Predicate)
+	{
+		FNode* Current = Head;
+		for (int CurrentIndex; CurrentIndex < Size; CurrentIndex++)
+		{
+			if (Predicate(Current->Element))
+			{
+				return *Current;
+			}
+			Current = Current->Next;
+		}
+		return nullptr;
+	}
+
+
+	template<typename Pred>
+	TDLList<T> FilterByPredicate(const Pred& Predicate)
+	{
+		TDLList<T> NewList;
+		FNode* Current = Head;
+		for (int CurrentIndex = 0; CurrentIndex < Size; CurrentIndex++)
+		{
+			if (Predicate(Current->Element))
+			{
+				NewList.AddTail(Current->Element);
+			}
+			Current = Current->Next;
+		}
+		return NewList;
+	}
+
+	template<typename Pred>
+	void RemoveAllByPredicate(const Pred& Predicate)
+	{
+		FNode* Current = Head;
+		for (int CurrentIndex = 0; CurrentIndex < Size; CurrentIndex++)
+		{
+			T Element = Current->Element;
+			Current = Current->Next;
+			if (Predicate(Current->Element))
+			{
+				Remove(CurrentIndex);
+				CurrentIndex--;
+			}
+		}
+	}
+
+
+
+
+
+	class FIterator
+	{
+	private:
+		FNode* Current = nullptr;
+	public:
+		FIterator(FNode* Currently)
+		{
+			Current = Currently;
+		}
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+
+			return *this;
+		}
+
+		T& operator*()
+		{
+			return Current->Element;
+		}
+
+		const T& operator*() const
+		{
+			return Current->Element;
+		}
+
+		bool operator!=(FIterator CurrentOther)
```

Comment:
An Iterator isn't something current, there is however an attribute of it that is Current, so the name of the other Iterator shouldn't be CurrentOther